lib/nostrum/struct/guild/role.ex

```diff
@@ -59,6 +60,9 @@ defmodule Nostrum.Struct.Guild.Role do
   @typedoc "Whether the role is mentionable"
   @type mentionable :: boolean
 
+  @typedoc "The hash of the role icon"
+  @type icon :: String.t()
```

Comment:
```suggestion
  @type icon :: String.t() | nil
```
One small thing since this field is nullable

Reply:
Nice catch!